app/scripts/services/fighter.js

```diff
@@ -0,0 +1,179 @@
+"use strict";
+
+function DiceRoller(options) {
+  // Default options and mechanics for the 1d20 strategy.
+  //
+  // May be overriden with an options hash provided as an argument to the dice
+  // roller.
+  var opts = _.extend({
+    difficulty: 10
+  }, options);
+  var _strategy = function() {
+    return _.random(1, 20);
+  };
+  var _criticalMax = 20;
+  var _criticalMin = 1;
+
+  return {
+    /**
+     * Set a specific rolling strategy.
+     *
+     * The default strategy is 1d20.
+     */
+    set strategy(strategyFn) {
+      _strategy = strategyFn;
+    },
+
+    /**
+     * Set a specific critical max threshold.
+     *
+     * The default threshold is 20 (max value for 1d20).
+     */
+    set criticalMax(threshold) {
+      _criticalMax = threshold;
+    },
+
+    /**
+     * Set a specific critical min threshold.
+     *
+     * The default threshold is 1 (min value for 1d20).
+     */
+    set criticalMin(threshold) {
+      _criticalMin = threshold;
+    },
+
+    /**
+     * Let's roll! Hopefully a good one.
+     *
+     * Access the value property of the roll to get the result.
+     *
+     * @param {object} stats: any custom stats used for computations
+     * @return {object} describes the roll
+     */
+    roll: function(stats) {
+      //this.value = _strategy();
+      // HAHA! Let's do only crits for now!!!
+      this.value = 20;
+      return {
+        value: this.value,
+        isSuccess: this.isSuccess(stats.difficulty),
+        isCritical: this.isCriticalSuccess(stats.criticalMax) || this.isCriticalFailure(stats.criticalMin)
+      }
+    },
+
+    /**
+     * Determine whether the roll is to be considered a success or a failure.
+     *
+     * @param {integer} difficulty: the score to beat to be successful.
+     * @return {boolean} true if the action succeed, false otherwise.
+     */
+    isSuccess: function(difficulty) {
+      return this.value > (difficulty || opts.difficulty);
+    },
+
+    /**
+     * Determine if an action is a critical success.
+     * Happens on a 20 - or lower with some weapons.
+     *
+     * @param {integer} threshold: custom value to overcome to trigger a critical success.
+     * @return {boolean} true if the action is a critical success, false otherwise.
+     */
+    isCriticalSuccess: function(threshold) {
+      return this.value >= (threshold || _criticalMax);
+    },
+
+    /**
+     * Determine if an action is a critical failure.
+     * Happens on a 1.
+     *
+     * @return {boolean} true if the action is a critical failure, false otherwise.
+     */
+    isCriticalFailure: function() {
+      return this.value === _criticalMin;
+    }
+  }
+}
+
+/**
+ * @ngdoc function
+ * @name rpgApp.service:Fighter
+ * @description
+ * Represents an entity taking part in a fight.
+ */
+
+function Fighter(fighter) {
+  var _target = null;
+
+  function onAttackSucceded(roll) {
+    // Monsters should have a weapon and attributes, like any fighters.
```

Comment:
About that comment: although I would definitely make the mobs behave the same as the player, one could argue it's not a good design. In that case, one could create some specialized Fighter constructors, customized using the prototype chain and overriding only what's different (like, the way damages are being computed—just a matter of extracting that piece of code in a function).